### PR TITLE
Bring in Azure ARM command

### PR DESCRIPTION
**Pull request by johnsimons**
Server PR - https://github.com/OctopusDeploy/OctopusDeploy/pull/6565
Calamari PR - https://github.com/OctopusDeploy/Calamari/pull/645

### CODE REVIEW

File: source/Calamari.AzureResourceGroup/Program.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        {
+        }
+
+        protected override void ConfigureContainer(ContainerBuilder builder, CommonOptions options)
+        {
+            base.ConfigureContainer(builder, options);
+
+            builder.RegisterType<TemplateService>();
+            builder.RegisterType<ResourceGroupTemplateNormalizer>().As<IResourceGroupTemplateNormalizer>();
+            builder.RegisterType<AzureContextScriptWrapper>().As<IScriptWrapper>().SingleInstance();
```
Comment (flin-8) [inline source/Calamari.AzureResourceGroup/Program.cs:25]: Script wrapper registration no longer needed right? if we override the assemblies to register

File: source/Sashimi.AzureResourceGroup/AzureResourceGroupActionHandler.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    class AzureResourceGroupActionHandler : IActionHandlerWithAccount
+    {
+        public string Id => SpecialVariables.Action.AzureResourceGroup.ResourceGroupActionTypeName;
+        public string Name => "Deploy an Azure Resource Manager template";
+        public string Description => "Deploy an Azure Resource Manager (ARM) template.";
+        public string? Keywords => null;
+        public bool ShowInStepTemplatePickerUI => true;
+        public bool WhenInAChildStepRunInTheContextOfTheTargetMachine => false;
+        public bool CanRunOnDeploymentTarget => false;
+        public ActionHandlerCategory[] Categories => new[] { ActionHandlerCategory.BuiltInStep, AzureConstants.AzureActionHandlerCategory };
```
Comment (flin-8) [inline source/Sashimi.AzureResourceGroup/AzureResourceGroupActionHandler.cs:16]: Server has 3 categories on the definition `ActionHandlerCategory.BuiltInStep, ActionHandlerCategories.Azure, ActionHandlerCategory.Script`

File: source/Calamari.AzureResourceGroup/InternalsVisibleTo.cs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+using System.Runtime.CompilerServices;
+
+[assembly:InternalsVisibleTo("Calamari.AzureResourceGroup.Tests")]
```
Comment (flin-8) [inline source/Calamari.AzureResourceGroup/InternalsVisibleTo.cs:3]: These should be in a `Properties` folder according to @slewis74

File: source/Calamari.AzureResourceGroup/Program.cs (hunk 1 above)
Comment (johnsimons) [reply source/Calamari.AzureResourceGroup/Program.cs:25]: yes, when I merge the calamari side I'll get that fixed

File: source/Sashimi.AzureResourceGroup/AzureResourceGroupActionHandler.cs (hunk 2 above)
Comment (johnsimons) [reply source/Sashimi.AzureResourceGroup/AzureResourceGroupActionHandler.cs:16]: Fixed

**Comment by flin-8** [comment]
Should also copy the `OnlyExposeWhatIsNecessary.cs` test to `Sashimi.AzureResourceGroup.Tests`

File: source/Calamari.AzureResourceGroup/InternalsVisibleTo.cs (hunk 3 above)
Comment (johnsimons) [reply source/Calamari.AzureResourceGroup/InternalsVisibleTo.cs:3]: Fixed

**Pull request merged**